MTM-46453 Added requirements for all major cockpit features. (#532)
SoftwareAG/c8y-docs: content/users-guide/cockpit-bundle/using-widgets.md

Widgets can display maps, images, graphs, tables and other graphic representations of data. Widgets are useful to track information, for example on alarms, assets or applications, or provide maps, quick links and more in dashboards or reports.

{{< c8y-admon-req >}}
ROLES & PERMISSIONS:

- To view widgets within dashboards: READ permission for permission type "Inventory" or READ permission for "Inventory" in the inventory roles

> - To edit widgets: ADMIN permission for permission type "Inventory" or CHANGE permission for "Inventory" in the inventory roles

Reviewer: ```suggestion
- To edit a widget: ADMIN permission for permission type "Inventory" or CHANGE permission for "Inventory" in the inventory roles
```

Widgets can display maps, images, graphs, tables and other graphic representations of data. Widgets are useful to track information, for example on alarms, assets or applications, or provide maps, quick links and more in dashboards or reports.

{{< c8y-admon-req >}}
ROLES & PERMISSIONS:

- To view widgets within dashboards: READ permission for permission type "Inventory" or READ permission for the "Inventory" Inventory roles
- To edit widgets: ADMIN permission for permission type "Inventory" or CHANGE permission for the "Inventory" Inventory roles

> - To create a widget: CREATE permission for permission type "Inventory" or CHANGR permission for the "Inventory" Inventory roles

Reviewer 2: I guess it's ADMIN not create, as you edit a dashboard

Author: changed

> - To create a widget: CREATE permission for permission type "Inventory" or CHANGR permission for the "Inventory" Inventory roles

Reviewer 2: typo: `CHANGR`

Author: fixed

- To remove a widget: ADMIN permission for permission type "Inventory" or CHANGE permission for the "Inventory" Inventory roles

>   {{< /c8y-admon-req >}}

Reviewer 3: ```suggestion
{{< /c8y-admon-req >}}
```

{{< c8y-admon-info >}}
Some of the widget will require additional permissions in order to visualize the data which they display. E.g. The alarms widget will require READ permission for permission type "Alarms" in order to view all of the alarms.
{{< /c8y-admon-info >}}

{{< product-c8y-iot >}} provides preset widget types, for details see the [Widgets collection](#widgets-collection).

<a name="adding-widgets"></a>

### To add a widget to a dashboard or a report

> 1.  Click **Add widget** in the top menu bar or click the **Add widget** button on the main page (only available in case of an empty dashboard/report).

Reviewer 3: ```suggestion
1. Click **Add widget** in the top menu bar or click the **Add widget** button on the main page (only available in case of an empty dashboard/report).
```

> 2.  In the **Add widget** dialog, select a widget type.

Reviewer 3: ```suggestion
2. In the **Add widget** dialog, select a widget type.
```

<img src="/images/users-guide/cockpit/cockpit-widget-add.png" name="Add widget">

> 3.  Next, configure the widget. According to the selected widget type, different parameters may be specified under **Configuration**. For details on each widget type refer to [Widgets collection](#widgets-collection).

Reviewer 3: ```suggestion
3. Next, configure the widget. According to the selected widget type, different parameters may be specified under **Configuration**. For details on each widget type refer to [Widgets collection](#widgets-collection).
```

> 4.  In the **Appearence** tab, you can customize the content and header style for the widget individually, in the same way as specifying the layout of a [dashboard](#creating-dashboards).

Reviewer 3: ```suggestion
4. In the **Appearence** tab, you can customize the content and header style for the widget individually, in the same way as specifying the layout of a [dashboard](#creating-dashboards).
```
